src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/BinaryFieldMapper.java

```diff
@@ -38,6 +39,13 @@ public AttributeValue reverse(FieldMapping fieldMapping, AttributeValue qualifie
         return convertFromBinary(fieldMapping.getTarget().getType(), fieldValue.getValue());
     }
 
+    @Override
+    public Predicate<AttributeValue> createFilter() {
+        final Predicate<ByteBuffer> prefixFilter =
+            BinaryFieldPrefixFunction.INSTANCE.createFilter(mtContext.getContext(), virtualTableName);
+        return attributeValue -> prefixFilter.test(attributeValue.getB());
+    }
```

Comment:
`FieldMapper`s and `PrefixFunction`s now support creating filters, so we can push the predicate as low as possible (this is the main perf optimization).